core/src/main/java/hudson/cli/GroovyshCommand.java

```diff
@@ -79,6 +81,7 @@ protected int run() {
         }
 
         Groovysh shell = createShell(stdin, stdout, stderr);
+        ScriptListener.fireScriptEvent(commandLine.toString(), "CLI", User.current());
```

Comment:
This does not actually record the script submitted via the shell by the user. It records the command line arguments (probably none) once, before the user actually starts using the shell. So all this does is record _that_ someone (and who_ used the `groovysh` command, but nothing about the actual script commands.

Reply:
Something like https://github.com/daniel-beck/jenkins/commit/9590e638db4f16877895211fb2db726363f6f94b may do it.

I have not yet figured out how the Groovysh profile files work; those could potentially still be used to bypass logging.

Reply:
> This does not actually record the script submitted via the shell by the user. It records the command line arguments (probably none) once, before the user actually starts using the shell. So all this does is record _that_ someone (and who) used the `groovysh` command, but nothing about the actual script commands.



> Something like [daniel-beck@9590e63](https://github.com/daniel-beck/jenkins/commit/9590e638db4f16877895211fb2db726363f6f94b) may do it.
> 
> I have not yet figured out how the Groovysh profile files work; those could potentially still be used to bypass logging.

Very good find, thank you! I implemented your solution. 

Regarding the profiles: If I understand Groovy Profiles correctly (and my tests weren't completely wrong), the Groovy profile on the client-side would have no effect.